src/libs/Navigation/linkTo.ts

```diff
@@ -1,72 +1,77 @@
 import {getActionFromState} from '@react-navigation/core';
-import _ from 'lodash';
+import {NavigationAction, NavigationContainerRef, NavigationState, PartialState} from '@react-navigation/native';
+import {Writable} from 'type-fest';
 import CONST from '@src/CONST';
 import NAVIGATORS from '@src/NAVIGATORS';
+import {Route} from '@src/ROUTES';
 import getStateFromPath from './getStateFromPath';
 import getTopmostReportId from './getTopmostReportId';
 import linkingConfig from './linkingConfig';
+import {NavigationRoot, RootStackParamList, StackNavigationAction} from './types';
 
 /**
  * Motivation for this function is described in NAVIGATION.md
  *
- * @param {Object} action action generated by getActionFromState
- * @param {Object} state The root state
- * @returns {Object} minimalAction minimal action is the action that we should dispatch
+ * @param action action generated by getActionFromState
+ * @param state The root state
+ * @returns minimalAction minimal action is the action that we should dispatch
  */
-function getMinimalAction(action, state) {
-    let currentAction = action;
-    let currentState = state;
-    let currentTargetKey = null;
+function getMinimalAction(action: NavigationAction, state: NavigationState): Writable<NavigationAction> {
+    let currentAction: NavigationAction = action;
+    let currentState: NavigationState | PartialState<NavigationState> | undefined = state;
+    let currentTargetKey: string | undefined;
 
-    while (currentState.routes[currentState.index].name === currentAction.payload.name) {
-        if (!currentState.routes[currentState.index].state) {
+    while (currentAction?.payload && 'name' in currentAction.payload && currentState?.routes[currentState?.index ?? -1].name === currentAction?.payload?.name) {
+        if (!currentState?.routes[currentState?.index ?? -1].state) {
             break;
         }
 
-        currentState = currentState.routes[currentState.index].state;
+        currentState = currentState?.routes[currentState?.index ?? -1].state;
+        currentTargetKey = currentState?.key;
 
-        currentTargetKey = currentState.key;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const payload = currentAction.payload as Record<string, any>;
```

Comment:
So if we know how this object will look like why we are typing it as `Record<string, any>` ?

Reply:
Payload is typed with `object`, do you have a different idea how to type it? 🤔

Reply:
```
payload: {
  name: payload?.params?.screen,
  params: payload?.params?.params,
  path: payload?.params?.path,
},
```

Cant we at least type it like this

````
{
   params: {
     screen: unknown; // maybe string?
     params: unknown;
     path: string? 
   }
}

Reply:
Let me test it 👍